### PR TITLE
[feature/12] Add lessons query

### LINKED ISSUE #11: Implement the themes query resolver

### LINKED ISSUE #12: Implement the lessons query resolver

### PR DESCRIPTION
**Pull request by pirvudoru**
### What does it fix?

Closes #11
Closes #12 

This PR puts in place the Domain module
Extracts mock repositories
Adds real repositories for Lessons and Themes
Seeds dummy data in the database

### CODE REVIEW

File: TeacherWorkout.Domain/Lessons/GetLessons.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+            _repository = repository;
+        }
+        
+        public PaginatedResult<Lesson> Execute(Input input)
+        {
+            var filterList = new List<IFilter>();
+            if (input.ThemeId.HasValue)
+            {
+                filterList.Add(new ThemeFilter(input.ThemeId.Value));
+            }
```
Comment (radubuciuceanu) [inline TeacherWorkout.Domain/Lessons/GetLessons.cs]: Won't it be cleaner to have a complex graph type together with a complex model used to filter lessons? By following this approach we can also add more filters without modifying the query contract.

**Review by radubuciuceanu — CHANGES_REQUESTED** [review]
image
Should we also remove those folder from inside the csproj file?

**Review by radubuciuceanu — CHANGES_REQUESTED** [review]
Should we also remove the Api/WeatherForecast file?

**Review by radubuciuceanu — CHANGES_REQUESTED** [review]
As far as we are using feature folders for commands, should we also move the models into their related feature folder?

image

File: TeacherWorkout.Domain/Lessons/GetLessons.cs (hunk 1 above)
Comment (pirvudoru) [reply TeacherWorkout.Domain/Lessons/GetLessons.cs]: Ended up doing that :+1:

**Review by catileptic — APPROVED** [review]

**Pull request merged**